OpenRA.Mods.Common/Traits/Air/ReturnOnIdle.cs

```diff
@@ -1,68 +0,0 @@
-#region Copyright & License Information
-/*
- * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
- * This file is part of OpenRA, which is free software. It is made
- * available to you under the terms of the GNU General Public License
- * as published by the Free Software Foundation, either version 3 of
- * the License, or (at your option) any later version. For more
- * information, see COPYING.
- */
-#endregion
-
-using System.Linq;
-using OpenRA.Mods.Common.Activities;
-using OpenRA.Traits;
-
-namespace OpenRA.Mods.Common.Traits
-{
-	[Desc("Return to a player owned RearmActor. If none available, head back to base and circle over it.")]
-	public class ReturnOnIdleInfo : ITraitInfo, Requires<AircraftInfo>
-	{
-		public object Create(ActorInitializer init) { return new ReturnOnIdle(init.Self, this); }
-	}
-
-	public class ReturnOnIdle : INotifyIdle
-	{
-		readonly AircraftInfo aircraftInfo;
-
-		public ReturnOnIdle(Actor self, ReturnOnIdleInfo info)
-		{
-			aircraftInfo = self.Info.TraitInfo<AircraftInfo>();
-		}
-
-		void INotifyIdle.TickIdle(Actor self)
-		{
-			// We're on the ground, let's stay there.
-			if (self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length < aircraftInfo.MinAirborneAltitude)
-				return;
-
-			var resupplier = ReturnToBase.ChooseResupplier(self, true);
-			if (resupplier != null)
-				self.QueueActivity(new ReturnToBase(self, aircraftInfo.AbortOnResupply, resupplier));
-			else
-			{
-				// nowhere to land, pick something friendly and circle over it.
```

Comment:
We might want to integrate the code below that searches for backup locations into `ReturnToBase`. That was a TODO point anyway.